SparseGrids/tsgGridFourier.hpp

```diff
@@ -0,0 +1,91 @@
+
+#ifndef __TASMANIAN_SPARSE_GRID_FOURIER_HPP
+#define __TASMANIAN_SPARSE_GRID_FOURIER_HPP
+
+#include <cstdlib>
+#include <math.h>
+#include <complex.h>
+
+#include "tsgEnumerates.hpp"
+#include "tsgIndexSets.hpp"
+#include "tsgCoreOneDimensional.hpp"
+#include "tsgIndexManipulator.hpp"
+#include "tsgLinearSolvers.hpp"
+#include "tsgOneDimensionalWrapper.hpp"
+#include "tsgGridCore.hpp"
+
+#include "tsgAcceleratedDataStructures.hpp"
+
+namespace TasGrid{
+
+class GridFourier{
+public:
+    GridFourier();
+    GridFourier(const GridFourier &fourier);
+    ~GridFourier();
+
+    void makeGrid(int cnum_dimensions, int cnum_outputs, int depth, const int* anisotropic_weights = 0, const int* level_limits = 0);
+    void copyGrid(const GridFourier *fourier);
+
+    void setTensors(IndexSet* &tset, int cnum_outputs);
+    int* referenceExponents(const int levels[], const IndexSet *list);
+
+    int getNumDimensions() const;
+    int getNumOutputs() const;
+    TypeOneDRule getRule() const;
+
+    int getNumLoaded() const;
+    int getNumNeeded() const;
+    int getNumPoints() const; // returns the number of loaded points unless no points are loaded, then returns the number of needed points
+    
+    void loadNeededPoints(const double *vals);
+    const double* getLoadedValues() const;
+    
+    double* getLoadedPoints() const;
+    void getLoadedPoints(double *x) const;
+    double* getNeededPoints() const;
+    void getNeededPoints(double *x) const;
+    double* getPoints() const;
+    void getPoints(double *x) const; // returns the loaded points unless no points are loaded, then returns the needed points
+
+	void calculateFourierCoefficients();
+
+    std::complex<double>* getBasisFunctions(const double x[]) const;
+    void getBasisFunctions(const double x[], std::complex<double> weights[]) const;
+
+    void evaluate(const double x[], double y[]) const;
+    void evaluateBatch(const double x[], int num_x, double y[]) const;
+
+	void clearAccelerationData();
```

Comment:
* check indentation across all files, there are many places where tabs are used instead of spaces